src/bindings/python/src/compatibility/openvino/inference_engine/__init__.py

```diff
@@ -29,5 +31,11 @@
 
 from .ie_api import *
 
+warnings.warn(
+    message="This API is deprecated and will be removed in 2024.0 release.",
+    category=FutureWarning,
+    stacklevel=2,
+)
+
```

Comment:
do we need to mark all API entries explicitly with deprecated?

Reply:
I second this, at least main classes like Core or Model. Also warnings in old API opsets would be great as they are often used.

Overall main inits **should** propagate all warnings but one-time only. Adding another layers of warnings will only help with advertisement of switching to new API.